File: gherkin/java/src/test/java/io/cucumber/gherkin/ParserTest.java
package io.cucumber.gherkin;

import io.cucumber.messages.IdGenerator;
import io.cucumber.messages.types.GherkinDocument;
import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class ParserTest {

    @Test
    public void change_default_language() {
        TokenMatcher matcher = new TokenMatcher("no");
        IdGenerator idGenerator = new IncrementingIdGenerator();
        Parser<GherkinDocument> parser = new Parser<>(new GherkinDocumentBuilder(idGenerator, "test.feature"));

        GherkinDocument gherkinDocument = parser.parse("Egenskap: i18n support\n", matcher, "test.feature");
        assertEquals("no", gherkinDocument.getFeature().get().getLanguage());
    }
}

File: gherkin/dart/test/Parser_test.dart
import 'package:gherkin/ast.dart';
import 'package:gherkin/parser.dart';
import 'package:gherkin/language.dart';
import 'package:test/test.dart';

void main() {
  test('Change default language', () {
      var languages = loadGherkinLanguagesFromJsonAsset();
      var dialectProvider = GherkinDialectProvider(languages, 'no');
      var matcher = TokenMatcher(dialectProvider);
      var idGenerator = IdGenerator.incrementingGenerator;

      var builder = GherkinDocumentBuilder(idGenerator);

      var tokenScanner = StringTokenScanner('Egenskap: i18n support\n');

      var parser = Parser<GherkinDocument>(builder);

      var gherkinDocument = parser.parse(tokenScanner, matcher);

      expect('no', gherkinDocument.feature.language);
  });
}

